agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java

```diff
@@ -55,23 +54,22 @@ public LogDataMapper(
     this.telemetryInitializer = telemetryInitializer;
   }
 
-  public void map(LogData log, Consumer<TelemetryItem> consumer) {
-    String stack = log.getAttributes().get(SemanticAttributes.EXCEPTION_STACKTRACE);
+  public TelemetryItem map(LogData log, @Nullable String stack, @Nullable Long itemCount) {
     if (stack == null) {
-      consumer.accept(createMessageTelemetryItem(log));
```

Comment:
where does the default properties and tags get applied now with this change?

Reply:
instead of passing in consumer, now this returns the object and the caller passes it to the consumer